luigi/contrib/hadoop.py

```diff
@@ -523,7 +524,10 @@ def run_job(self, job, tracking_url_callback=None):
 
         job.dump(self.tmp_dir)
 
-        run_and_track_hadoop_job(arglist, tracking_url_callback=tracking_url_callback)
+        job_env = os.environ.copy()
+        if self.username:
+            job_env["HADOOP_USER_NAME"] = self.username
```

Comment:
Creating a copy of the env variables and updating the `job_env` won't actually update the environment itself:

```python
>>> import os
>>> job_env = os.environ.copy()
>>> job_env["HADOOP_USER_NAME"] = "bar"
>>> os.environ.get("HADOOP_USER_NAME", "not set")
'not set'
```

You'll need to update it directly:

```python
>>> os.environ["HADOOP_USER_NAME"] = "bar"
>>> os.environ.get("HADOOP_USER_NAME", "not set")
'bar'
```

Same with the other files below that try to set the env var.